libs/core/src/lib/flexible-column-layout/flexible-column-layout.component.ts

```diff
@@ -0,0 +1,421 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ContentChild,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  SimpleChanges,
+  TemplateRef
+  } from '@angular/core';
+  import { fromEvent, Subscription } from 'rxjs';
+  import { debounceTime } from 'rxjs/operators';
+  import {
+  ONE_COLUMN_START_FULL_SCREEN,
+  ONE_COLUMN_MID_FULL_SCREEN,
+  ONE_COLUMN_END_FULL_SCREEN,
+  TWO_COLUMNS_START_EXPANDED,
+  TWO_COLUMNS_MID_EXPANDED,
+  TWO_COLUMNS_END_EXPANDED,
+  THREE_COLUMNS_MID_EXPANDED,
+  THREE_COLUMNS_END_EXPANDED,
+  THREE_COLUMNS_START_MINIMIZED,
+  THREE_COLUMNS_END_MINIMIZED,
+  SM_SCREEN_SIZE,
+  MD_SCREEN_SIZE,
+  LG_SCREEN_SIZE,
+  XL_SCREEN_SIZE,
+  ScreenSize,
+  Layout,
+  ColumnSeparatorValue
+} from './constants';
+  
+@Component({
+  selector: 'fd-flexible-column-layout',
+  templateUrl: './flexible-column-layout.component.html',
+  styleUrls: ['./flexible-column-layout.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class FlexibleColumnLayoutComponent implements AfterViewInit, OnChanges, OnDestroy, OnInit {
+
+  /**
+   * the template that provides the content of the first/start/left column
+   */
+  @ContentChild('startColumn')
+  startColumn: TemplateRef<any>;
+
+  /**
+   * the template that provides the content of the middle column
+   */
+  @ContentChild('midColumn')
+  midColumn: TemplateRef<any>;
+
+  /**
+   * the template that provides the content of the last/end/right column
+   */
+  @ContentChild('endColumn')
+  endColumn: TemplateRef<any>;
+  
+  /**
+   * the background design of the component
+   * Options include: 'solid' | 'translucent' | 'transparent'
+   * The default is set to 'solid'
+   */
+  @Input()
+  backgroundDesign: 'solid' | 'translucent' | 'transparent' = 'solid';
+
+  /**
+   * the layout of the component
+   * Options include: 'OneColumnStartFullScreen' | 'OneColumnMidFullScreen' |
+   * 'OneColumnEndFullScreen' | 'TwoColumnsStartExpanded' | 'TwoColumnsMidExpanded' |
+   * 'TwoColumnsEndExpanded' | 'ThreeColumnsMidExpanded' | 'ThreeColumnsEndExpanded' |
+   * 'ThreeColumnsStartMinimized' | 'ThreeColumnsEndMinimized'
+   */
+  @Input()
+  layout: Layout = ONE_COLUMN_START_FULL_SCREEN;
+
+  /**
+   * the event emitted on layout value cchange
+   */
+  @Output()
+  layoutChange: EventEmitter<Layout> = new EventEmitter<Layout>();
+
+  /**
+   * user defined break point for MD screens
+   * the default (Fiori 3) value is 599
+   */
+  @Input()
+  mdBreakPoint = 599;
+
+  /**
+   * user defined break point for LG screens
+   * the default (Fiori 3) value is 1023
+   */
+  @Input()
+  lgBreakPoint = 1023;
+
+  /**
+   * user defined break point for XL screens
+   * the default (Fiori 3) value is 1439
+   */
+  @Input()
+  xlBreakPoint = 1439;
```

Comment:
Do you need `xlBreakPoint` there? I can see that on docs and in code it shares behaviour with `lgBreakPoint`, so IMO only lg shoudl be kept

Reply:
removed. The idea was to have it in case the user passed a different break point for XL